examples/zynq-mpsoc/ad08-streams.cpp

```diff
@@ -0,0 +1,262 @@
+/*
+ * See LICENSE for more information about licensing
+ *
+ * Copyright 2024
+ * Author: Luis G. Leon-Vega <luis.leon@ieee.org>
+ */
+
+// clang-format off
+#include <algorithm>
+#include <cstdint>
+#include <cynq/accelerator.hpp>
+#include <cynq/datamover.hpp>
+#include <cynq/execution-graph.hpp>
+#include <cynq/hardware.hpp>
+#include <cynq/memory.hpp>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <third-party/timer.hpp>
+// clang-format on
+
+/**
+ * @example zynq-mpsoc/ad08-streams.cpp
+ *
+ * This is a use case that involves two accelerators with AXI-MM ports.
+ * The accelerators are executed in parallel using IExecutionGraph
+ */
+
+/*
+ * Running: sudo ./builddir/examples/ad08-kria 4 4 4
+ *
+ * Result:
+ *   MatMul Result:
+ *   728 1379 2026 2674 3322 3970 4618 5267 787 1483 2181 2879 3578 4276 4975
+ *   5672 842 1589 2338 3087 3835 4583 5330 6078 898 1696 2495 3292 4091 4888
+ *   5686 6486
+ *   ElementWise Result:
+ *   1057 1077 1097 1117 1137 1157 1177 1197 1217 1237 1257 1277 1297 1317
+ *   1337 1357 1377 1397 1417 1437 1457 1477 1497 1517 1537 1557 1577 1597
+ *   1617 1637 1657 1677
+ */
+
+typedef union {
+  uint16_t rvalues[4];
+  uint64_t packet;
+} PacketU;
+
+using DataType = uint16_t;
+using namespace cynq;  // NOLINT
+
+// Given by the example
+static constexpr char kBitstream[] = AD08_BITSTREAM_LOCATION;
+
+// Given by the design
+static constexpr uint64_t kElemWiseAddr = 0xA0000000;
+static constexpr uint64_t kMatMulAddr = 0xA0020000;
+static constexpr uint64_t kDmaAddress = 0xA0010000;
+#define XMATMUL_CONTROL_ADDR_A_DATA 0x10
+#define XMATMUL_CONTROL_ADDR_B_DATA 0x1c
+#define XMATMUL_CONTROL_ADDR_C_DATA 0x28
+#define XMATMUL_CONTROL_ADDR_A_ROWS_DATA 0x34
+#define XMATMUL_CONTROL_ADDR_B_COLS_DATA 0x3c
+#define XMATMUL_CONTROL_ADDR_C_COLS_DATA 0x44
+#define XELEMENTWISE_CONTROL_ADDR_IN1_DATA 0x10
+#define XELEMENTWISE_CONTROL_ADDR_IN2_DATA 0x1c
+#define XELEMENTWISE_CONTROL_ADDR_OUT_R_DATA 0x28
+#define XELEMENTWISE_CONTROL_ADDR_SIZE_DATA 0x34
+#define XELEMENTWISE_CONTROL_ADDR_OP_DATA 0x3c
+
+/* Auxiliary function to fill buffers */
+static void fill_data(std::shared_ptr<IMemory> buffer, const size_t num,
+                      const DataType start_value = 100,
+                      const DataType step_value = 10) {
+  DataType* data = buffer->HostAddress<DataType>().get();
+  for (size_t i = 0; i < num; ++i) {
+    data[i] = static_cast<DataType>(start_value + step_value * i);
+  }
+}
+
+/* Auxiliary function to print */
+static void print_data(std::shared_ptr<IMemory> buffer, const size_t num) {
```

Comment:
now that I see this functions repeated here, maybe the same comment given for CYNQ_INFO might apply here.

Reply:
Applied